.github/workflows/build.yml

```diff
@@ -0,0 +1,114 @@
+on:
+  push:
+  pull_request:
+
+name: 🏗️ Build
+
+jobs:
+  build-arch-server:
+    runs-on: ubuntu-latest
+    container: archlinux:base-devel
+
+    steps:
+      - name: 📥 Checkout
+        uses: actions/checkout@v3
+
+      - name: 👷 Build Dependencies
+        run: "pacman --noconfirm -Syu base-devel cmake gcc git make pipewire pipewire-pulse"
+
+      - name: 🔨 Build
+        run: |
+          cmake -B build
+          cmake --build build
+
+      - name: 🚀 Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: server-arch
+          path: build/server
+
+  build-arch-addon:
+    runs-on: ubuntu-latest
+    container: archlinux:base-devel
+
+    steps:
+      - name: 📥 Checkout
+        uses: actions/checkout@v3
+
+      - name: 👷 Build Dependencies
+        run: "pacman --noconfirm -Syu base-devel cmake gcc git make pipewire pipewire-pulse"
+
+      - name: 🍃 Install Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: 🍃 Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+          run_install: false
+
+      - name: 🔨 Build
+        run: |
+          pnpm install
+
+      - name: 🚀 Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: addon-arch
+          path: build/Release
+
+  build-fedora-server:
+    runs-on: ubuntu-latest
+    container: fedora:latest
+
+    steps:
+      - name: 📥 Checkout
+        uses: actions/checkout@v3
+
+      - name: 👷 Build Dependencies
+        run: "dnf install -y pipewire-devel make automake git gcc gcc-c++ kernel-devel cmake"
+
+      - name: 🔨 Build
+        run: |
+          cmake -B build
+          cmake --build build
+
+      - name: 🚀 Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: server-fedora
+          path: build/server
+
+  build-fedora-addon:
+    runs-on: ubuntu-latest
+    container: fedora:latest
+
+    steps:
+      - name: 📥 Checkout
+        uses: actions/checkout@v3
+
+      - name: 👷 Build Dependencies
+        run: "dnf install -y pipewire-devel make automake git gcc gcc-c++ kernel-devel cmake"
+
+      - name: 🍃 Install Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: 🍃 Install pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+          run_install: false
+
+      - name: 🔨 Build
+        run: |
+          pnpm install
+
+      - name: 🚀 Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: addon-fedora
+          path: build/Release
```

Comment:
Compiling for arch is to ensure that no compiler update suddenly breaks the build (i.e. there were some permissive changes some time ago that would cause breakage with certain build flags)

And compiling for fedora is to ensure that we have a build with a glibc version that isn't necessarily the latest.

I chose fedora as ubuntu doesn't use pipewire by default and from experience most ubuntu users won't explicitly install it either - fedora seems to be widely used and ships pipewire by default so that's most likely the biggest target audience so having a build based fedora:latest' glibc version should suffice and we probably don't need builds for older glibc versions

Reply:
Also, the ubuntu-lts GCC Version is so ancient that it doesn't even support C++20 if I'm not mistaken

Reply:
Are you sure the Arch build works on the Steam Deck, then? I can test it if you like.

Reply:
> Are you sure the Arch build works on the Steam Deck, then? I can test it if you like.

It should - if it doesn't the Fedora Build will work for sure but it would be great if you could test it ^^!

Reply:
@D3SOX Were you able to test?

Reply:
I'll resolve this for now, let me know if there's anything we should still discuss / test ^^